scantool/diag_tty_unix.c

```diff
@@ -1316,12 +1316,10 @@ char **diag_tty_getportlist(int *numports) {
 					continue;
 				}
 
-				char **templist = strlist_add(portlist, ffn, elems);
-				if (!templist) {
-					strlist_free(portlist, elems);
```

Comment:
portlist is not a valid pointer after being given to strlist_add, because of the realloc in strlist_add.